### ISSUE TITLE
[Feature Request] Scheduled sync

### ISSUE THREAD

**Issue by TTTPOB** [x0b/rcx#38]
The reason why I need this feature is that I've scheduled some backup tasks on my phone but I want them to be synced to cloud.

**Comment by lllluke** [comment]
I would like to have that feature, too.
It would also be usable as a camera upload function.

**Comment by newhinton** [comment]
I am currently implementing (#37) a way to start a specific sync via tasks. There will be a service which exposes the ability to start a task via an intent. However, i do not think that this app is going to provide the ability to set a time where a task should run (This is my opinion, it may be that the maintainer has a different opinion on that matter)

For now the pull request allows arbitrary apps to start a task, so tasker or some other similar app could help you achieve your goal

**Comment by x0b** [comment]
>  However, i do not think that this app is going to provide the ability to set a time where a task should run (This is my opinion, it may be that the maintainer has a different opinion on that matter)


It actually sounds like a great idea. There might be technical challenges though since Android vendors [like to kill background processes](https://dontkillmyapp.com/), which are probably required to run a task scheduler.

**Comment by newhinton** [comment]
@x0b That is quiet easily solved, i wrote multiple apps that use the Alarmmanager to schedule tasks (including an app that triggers an rclone task ;) )

I have found that this works quiet well and i use it for multiple things.
If you want to try it out:
https://git.felixnuesse.de/Felix/timed-intent-trigger/

**Comment by x0b** [comment]
Yeah, I was probably more concerned by the prospect of an onslaught of issues from people where the phone kills the alarms. If it works, its fine.

**Comment by ioogithub** [comment]
I am currently using RCX to backup phone data as well. Right now it is a very manual process. A way to automatically or schedule a backup would be a killer feature for me.

**Comment by dmgr** [comment]
Might be worth looking at the logic implemented in a Syncthing scanning feature:
https://docs.syncthing.net/users/syncing.html#scanning

> There are two methods how Syncthing detects changes: By regular full scans and by notifications received from the filesystem (“watcher”).

**Comment by dmgr** [comment]
This issue should have not been closed, TTTPOB...

**Comment by TTTPOB** [comment]
> This issue should have not been closed, TTTPOB...

sorry for that, you are right, this is a public feature request now

**Comment by kojid0** [comment]
What about a 1x1 sync widget for a first fast implementation?

**Comment by newhinton** [comment]
@kojid0 The sync feature is already done, it only needs reviewing. If you want to try it, i can upload an apk ([or you build it yourself](https://github.com/x0b/rcx/pull/84))

**Comment by kojid0** [comment]
Didn't saw that. That's awesome! Can you share an apk, please?

**Comment by newhinton** [comment]
[app-oss-arm64-v8a-debug.zip](https://github.com/x0b/rcx/files/7837759/app-oss-arm64-v8a-debug.zip)


Use at your own risk. I am using this now for more than two years, and it works great though

**Comment by kojid0** [comment]
> Use at your own risk. I am using this now for more than two years, and it works great though

Thank you, will try it!

**Comment by newhinton** [comment]
@kojid0 Please report any issues you have here: #84

**Comment by ms866** [comment]
> [app-oss-arm64-v8a-debug.zip](https://github.com/x0b/rcx/files/7837759/app-oss-arm64-v8a-debug.zip)
> 
> Use at your own risk. I am using this now for more than two years, and it works great though

@newhinton I tried installing that APK but running into "App Not Installed" Error

Enabled "Install from unknown sources"
Disabled "Google Play Protect"
Running Android 11

Any idea what might be causing the error?

**Comment by newhinton** [comment]
@ms866 You may have an arm7 phone, what brand&model do you have?

**Comment by ms866** [comment]
@newhinton 
OnePlus8Pro
Qualcomm Snapdragon 865
 Kryo 585 CPU (64-bit)
Adreno 650 GPU

**Comment by newhinton** [comment]
@ms866 Hmm, that should work, its the appropriate architecture. 

 I have attached the universal-apk, you could try that. 

 However: That specific build is brand new hot out of the oven and there are some things broken in it ;) Nothing major though, im implementing a retry button. 
 
 
Edit: Zip is missing because not uploading. Give me a sec
Edit2: Okay github does not allow files above 25m
[app-oss-armeabi-v7a-debug.zip](https://github.com/x0b/rcx/files/8041943/app-oss-armeabi-v7a-debug.zip)
[app-oss-arm64-v8a-debug.zip](https://github.com/x0b/rcx/files/8041946/app-oss-arm64-v8a-debug.zip)

I have now attached apks from that build, just not the universal one. Though i guess the v7a version will not work, you can try it anyway

**Comment by ms866** [comment]
@newhinton 
I was just so pumped when I saw you mention the scheduling feature
I could literally get rid of syncthing if so!

**Comment by newhinton** [comment]
@ms866 and i am sure we will get it working! Worst case: we will set you up with android studio and run it manually!

**Comment by ms866** [comment]
@newhinton 
Yeah 8a gave me the same error :/

**Comment by newhinton** [comment]
both?
Do you want to try setting it up yourself?

**Comment by ms866** [comment]
@newhinton 
7a gave me an error when extracting it to begin with
I would have to wait till the weekend to set it up myself
Thanks for all the help and patience
Also let me know if this conversation is a bit too long to be had on this github issue. I'm more than happy to continue this elsewhere

**Comment by wokawoka** [comment]
any news on this?

**Comment by newhinton** [comment]
@wokawoka Please see https://github.com/newhinton/extRact

**Comment by wokawoka** [comment]
@newhinton it looks great, thanks!
I'm going to test it right now
